packages/lib/src/treeUtils/sandbox.ts

```diff
@@ -15,7 +15,7 @@ import { assertIsFunction, failure } from "../utils"
 /**
  * Callback function for `SandboxManager.withSandbox`.
  */
-export type WithSandboxCallback<T extends object, R> = (
+export type WithSandboxCallback<T extends object | [object], R> = (
```

Comment:
is this needed? after all any tuple is an array, which is an object in itself.
Also [object] is a tuple of length 1 with a single object (unlike object[])

Reply:
I think so, although I see your point:

https://www.typescriptlang.org/play/#code/GYVwdgxgLglg9mABMAPAFUQUwB5U2AEwGdE4AjAK02gD4AKbALkTQEpmMBvAKET8QBOmKCAFJs3AL7duEADYBDIiQCCiTomnylJAELrNsxcsQBhA9NkIiURAsQBeRGEwB3RCrqsrYG4jKOzm6Iul4+fhCBLu6mYdzAdAreCWTJdBBpANoKADT+ALpZuf55EIVAA

Here, I removed the extra `| [object]` and type inference yields an array of a union type.

Reply:
Oh well, mysteries of TS :) Whatever works works I guess :D

Just bear in mind that the tuple [5, a] // [number, object] seems to be valid as well (maybe because an array is an object?)

Also it seems to be equivalent to `object | []`

This reminds me, what will happen when the original node is actually an array?

Reply:
529f047 adds support for passing an array node to `withSandbox` which is treated as a single node in terms of obtaining the sandbox copy.